src/app/student/events/page.tsx

```diff
@@ -0,0 +1,19 @@
+import { EventsTimeline } from "@/app/student/_components/EventsTimeLine"
+import { Page } from "@/components/shared/Page"
+import { fetchEvents } from "@/components/shared/hooks/api/useEvents"
+import { Suspense } from "react"
+
```

Comment:
In order to make this page statically generated we need to implement [generateStaticParams](https://nextjs.org/docs/app/api-reference/functions/generate-static-params), this will help next to load all the events during build time, I think a good revalidation time here is quite short, maybe rebuild once every hour, to fix the revalidation I think you need to pass the following to the settings in 
```ts
fetchEvents({
    next: {
        revalidate: 60*60 // Once every hour
    }
})
```